src/containers/tud/TimeUseDiaryContainer.js

```diff
@@ -140,6 +161,32 @@ const TimeUseDiaryContainer = () => {
     setFormData(currFormData);
   };
 
+  const changeLanguage = (lng :Object) => {
+    if (lng !== null) {
+      i18n.changeLanguage(lng.value);
+      setSelectedLanguage(lng);
+    }
+  };
+
+  const onConfirmChangeLanguage = () => {
+    if (languageToChangeTo !== null) {
+      setChangeLanguageModalVisible(false);
+      setShouldReset(true);
+      changeLanguage(languageToChangeTo);
+    }
```

Comment:
This looks like a case for `useReducer`. Not a blocker for merging, though I recommend looking into it as a follow-up.